### PR TITLE
GovTool user stories

### PR DESCRIPTION
**Pull request by spannercode**
This PR includes the user stories covered in GovTool for which automated tests are written (or to be written).

### CODE REVIEW

File: userStoryInventoryChangHF.md [hunk 1]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
+- Then if I have multiple wallets (CIP95 compatibles & non-compatibles) only CIP95 compatible wallets are listed and I will be asked which wallet to connect. Once I choose the wallet and give permission to the wallet to connect, then the app will connect to the wallet.
+
+## 2A: Show Delegated dRep ID
+
+**User Story:** As an Ada holder or dRep, I want to be able to see the dRep ID of the dRep that I have delegated to so that I know that I have delegated to someone and who I have delegated to. I also know that the delegation has been successful.
+
+### Acceptance Criteria
+
+- Given that I have connected my wallet to the app and I have delegated to a dRep.
+- When I look in the top right of my screen on any page.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:99]:
```suggestion
- When I look in the top right of my screen on any page.
```
top right correct?

File: userStoryInventoryChangHF.md [hunk 2]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
+
+### Acceptance Criteria
+
+- Given that I do not have a compatible wallet connected to GovTool.
+- When I attempt to visit the URL of the Delegate to DRep page.
+- Then I can see the delegation options.
+
+## 2L: Delegate to Self for Registered DReps
+
+**User Story:** As a registered DRep connected to GovTool with my wallet, I want to be able to delegate to myself via my connected wallet on the Delegate to DRep page.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:215]: Currently on GovTool, when a user registers to be a DRep, GovTool forces them to delegate to themself in the same transaction

File: userStoryInventoryChangHF.md [hunk 3]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
+## 3D: Registering as a dRep on Connected Wallet State
+
+**User Story:** As a dRep (someone who wants to be a dRep) who has connected their wallet to the app, I want to be able to register as a dRep so that I can fulfill the duties of a dRep.
+
+### Acceptance Criteria
+
+- Given that I have logged into the app.
+- When I visit the dRep registration page.
+- Then I should be presented with a registration form consisting of:
+  - An input box for my dRep metadata URL.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:304]: what about the metadata form we have currently?

File: userStoryInventoryChangHF.md [hunk 4]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
+
+### Acceptance Criteria
+
+- Given that I am on the dRep registration page.
+- If/when I have entered some information into the input box for metadata URL, then it will be checked against the following criteria:
+  - It must be ≤ 64 bytes in length.
+  - It must be UTF-8 compatible OR
+  - It must be empty.
+- And if/when I have entered some information into the input box for the hash of my metadata URL, then it should be checked against the following criteria:
+  - It must be 32 bytes in length.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:320]:
```suggestion
  - It must be at most 128 bytes in length.
```
https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/conway/impl/cddl-files/conway.cddl#L377C1-L377C26

File: userStoryInventoryChangHF.md [hunk 5]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
+
+### Acceptance Criteria
+
+- Given that I have chosen how to vote on the UI of a governance action's details.
+- When I add a metadata anchor to the UI and click the vote button.
+- Then the resulting transaction will include my metadata anchor.
+
+## 5A: Required Information Displayed for Governance Action
+
+**User Story:** As any user, when I look at a governance action it will have the following information displayed so that I can make an informed choice.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:473]: what information?

File: userStoryInventoryChangHF.md [hunk 6]
```diff
@@ -39,41 +39,646 @@ This section shows the areas or sources thus far
...
 
+## 5D: Reason URL and Reason Hash are Valid
 
+**User Story:** As a dRep, I want the data I have input into the reason URL box and reason hash input box to be valid so that my data is readable once it has been stored on the chain.
+
+### Acceptance Criteria
+
+- Given that I am in the process of casting my vote on a governance action.
+- If/when I have entered some information into the reason URL input box then it will be checked against the following criteria:
+  - It must be ≤ 64 bytes in length.
```
Comment (Ryun1) [inline userStoryInventoryChangHF.md:513]:
```suggestion
  - It must be ≤ 128 bytes in length.
```

**Review by Ryun1 — APPROVED** [review]
Awesome work!

See, minor comments from me :)

File: userStoryInventoryChangHF.md (hunk 1 above)
Comment (NabinKawan) [reply userStoryInventoryChangHF.md]: It has been updated.

File: userStoryInventoryChangHF.md (hunk 3 above)
Comment (NabinKawan) [reply userStoryInventoryChangHF.md]: This has been updated with the latest test.

File: userStoryInventoryChangHF.md (hunk 5 above)
Comment (NabinKawan) [reply userStoryInventoryChangHF.md]: Thanks, The information has been included.

File: userStoryInventoryChangHF.md (hunk 6 above)
Comment (NabinKawan) [reply userStoryInventoryChangHF.md:603]: It has been updated.

**Comment by Ryun1** [comment]
Thanks @NabinKawan for [f4922db](https://github.com/IntersectMBO/cardano-test-plans/pull/37/commits/f4922db9fc027cf95e3c5563d5290216985a2f07)
Can you resolve the merge conflicts, then we can merge this PR 💪

**Comment by NabinKawan** [comment]
Hi @Ryun1, Sorry for that. The merge conflicts has been resolved now.

**Review by carloslodelar — APPROVED** [review]

**Review by CarlosLopezDeLara — APPROVED** [review]

**Pull request merged**